src/frontend/tsx/Settings.tsx

```diff
@@ -1,10 +1,84 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { pieceThemes, boardThemes } from '../ts/helpers/themes'
 
-export function Settings () {
+// XX: Visuals. If a light and dark mode is implemented. Add logic to change
+// the black/white for each piece.
+//
+// XX: Add grid-area names to these list items. If grid is what we're using.
+//        <ul className="theme-list">
+//          {
+//            pieceThemes.map((theme, ind) => {
+//              let area = String.fromCharCode(ind + 97);
+//              return (
+//                <li key={ind} className={`theme game-icon ${theme}`}>
+//                  <piece className="game-icon black knight" style={`grid-area: ${area}`}/>
+//                </li>
+//              )
+//            })
+//          }
+//        </ul>
+//
+export function Settings ({setPieceTheme, setBoardTheme}: any) {
+  const [prevPieceTheme, setPrevPieceTheme] = useState<string>('empty')
+  const [prevBoardTheme, setPrevBoardTheme] = useState<string>('empty')
   return (
     <div className='settings-container'>
       <div id="visuals-settings" className="control-panel-container col">
         <h4 className="control-panel-header">Visuals</h4>
+        <ul className="theme-list">
+          {
+            pieceThemes.map((theme, ind) => {
+              const handleClick = () => {
+                setPieceTheme(`${theme}`)
+                toggleSelect()
+                setPrevPieceTheme(`${theme}`)
+                console.log(prevPieceTheme)
+              }
+
+              function toggleSelect () {
+                document.getElementById(theme).classList.toggle('selected')
+                document.getElementById(prevPieceTheme).classList.toggle('selected')
+              }
+
+              return (
+                <li key={ind}
+                    id={theme}
+                    className={`theme ${theme}`}
+                    onClick={() => handleClick()}
+                >
+                  <piece className="theme-icon black knight"/>
+                </li>
```

Comment:
After combining `state` with `localStorage`, you shouldn't need to use `toggleSelect`. Just calling `setPieceTheme` should be enough, and the interface will rerender because the state of `localStorage` will change. Thus, you should be able to do something like this:
```
pieceThemes.map((theme, ind) => {
  
  status = (theme === localStorage.getItem('pieceTheme')) ? 'selected' : 'unselected'

  const handleClick = () => {
    setPieceTheme(`${theme}`)
  }

  return (
    <li key={ind}
        id={theme}
        className={`theme ${theme} ${status}`}
        onClick={() => handleClick()}
    >
      <piece className="theme-icon black knight"/>
    </li>)})
```